aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -329,10 +329,8 @@ std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTab
   // to let the original CompositeImplicitAutograd handle Undefined
   if (dispatch_key != DispatchKey::Undefined && isIncludedInAlias(dispatch_key, DispatchKey::CompositeImplicitAutogradNestedTensor)) {
     if (auto nested_registration = getKernelForDispatchKey(DispatchKey::CompositeImplicitAutogradNestedTensor)) {
-      if (!has_backend_kernel) {
```

Comment:
Just to close the loop on this from our offline conversation - I double checked, and the idea of line 341 (where this was copied from is that if you have a table that looks like this:
```
dispatch:
  XLA: xla_kernel
  CompositeImplicitAutograd: composite_kernel
```

Then what the dispatcher does in the above case is that:
```
XLA key: xla_kernel is registered
AutogradXLA key: **nothing** is registered
```

The idea there is that we actually don't want to register the composite kernel to the autograd key - otherwise, we would end up running that decomposition, and end up completely skipping the XLA kernel that someone explicitly registered. Instead, if you try to to run the op normally, you'll get an error correctly saying that no autograd kernel exists (which is required, because we defined a kernel for the forward).

This has only really been important for when we have a decomposition for an op in core, and an external backend (like XLA) wants to write their own custom forward and custom backward. So this probably won't matter to nested tensor unless that becomes a real use case (say, XLA wants to write a custom forward + backward for nested tensor's reshape). That seems... like something that won't happen any time soon?

Reply:
If XLA starts contributing to NestedTensor that would be awesome lol but I think you are right and that doesn't seem like its on the immediate horizon.

Also just to clarify:
``` yaml
dispatch:
  XLA: xla_kernel
  CompositeImplicitAutogradNestedTensor: composite_kernel
  ```
Would still produce
  ```
XLA key: xla_kernel is registered
AutogradXLA key: **nothing** is registered
```

Because CompositeImplicitAutogradNestedTensor only effects the nestedtensor keys

Reply:
yep!